pygfx/geometries/_text.py

```diff
@@ -576,65 +630,234 @@ def _apply_layout(self):
         # Prepare
 
         font_size = self._font_size
+        # We try to follow CSS  which multiplies the line_height by the font_size as well
+        line_height = self.line_height * font_size
+
         anchor = self._anchor
+        text_align = self._text_align
+        text_align_last = self._text_align_last
+        if text_align_last == "auto":
+            if text_align == "justify":
+                text_align_last = "start"
+            elif text_align == "justify-all":
+                text_align_last = "justify"
+            else:
+                text_align_last = text_align
+
+        if text_align == "justify-all":
+            text_align = "justify"
+
+        # TODO: handle Right to Left (RTL) text
+        if self._direction == "ltr":
+            if text_align == "start":
+                text_align = "left"
+            elif text_align == "end":
+                text_align = "right"
+
+            if text_align_last == "start":
+                text_align_last = "left"
+            elif text_align_last == "end":
+                text_align_last = "right"
+        elif self._direction == "rtl":
+            if text_align == "end":
+                text_align = "left"
+            elif text_align == "start":
+                text_align = "right"
+
+            if text_align_last == "end":
+                text_align_last = "left"
+            elif text_align_last == "start":
+                text_align_last = "right"
+
         positions_array = self.positions.data
         sizes_array = self.sizes.data
         is_horizontal = self._direction is None or self._direction in ("ltr", "rtl")
 
-        left = right = 0
-        top = bottom = 0
+        # The algorightm doesn't support text alignment for ttb and btt yet
+        if not is_horizontal:
+            text_align = "left"
+            text_align_last = "left"
```

Comment:
I realized that center alignment was causing more bugs on the ttb alignment.

Somebody with enough determination should go through the alignments correctly, but for now, this helps things be a little more "correct" than just "wrong".

Reply:
👍